engine/src/main/java/com/redhat/cloud/notifications/processors/email/EmailSubscriptionTypeProcessor.java

```diff
@@ -227,53 +318,62 @@ private void processAggregateEmailsByAggregationKey(EmailAggregationKey aggregat
 
                     Context.ContextBuilder contextBuilder = new Context.ContextBuilder();
                     aggregation.getKey().forEach(contextBuilder::withAdditionalProperty);
-
-                    Action action = new Action();
                     action.setContext(contextBuilder.build());
-                    action.setEvents(List.of());
-                    action.setOrgId(aggregationKey.getOrgId());
-                    action.setApplication(aggregationKey.getApplication());
-                    action.setBundle(aggregationKey.getBundle());
-
-                    // We don't have an event type as this aggregates over multiple event types
-                    action.setEventType(null);
-                    action.setTimestamp(LocalDateTime.now(ZoneOffset.UTC));
-
-                    Event event = new Event();
-                    event.setId(UUID.randomUUID());
                     event.setEventWrapper(new EventWrapperAction(action));
 
-                    emailSender.sendEmail(aggregation.getValue(), event, subject, body, false);
+                    NotificationHistory history = emailSender.sendEmail(aggregation.getValue(), event, subject, body, false, endpoint);
+                    Integer totalRecipients = (Integer) history.getDetails().get(TOTAL_RECIPIENTS_KEY);
+                    if (NotificationStatus.SUCCESS == history.getStatus()) {
+                        nbRecipientsSuccessfullySent += totalRecipients;
+                    } else {
+                        nbRecipientsFailureSent += totalRecipients;
+                    }
```

Comment:
`history` can be `null`.
```suggestion
                    if (history != null) {
                        Integer totalRecipients = (Integer) history.getDetails().get(TOTAL_RECIPIENTS_KEY);
                        if (NotificationStatus.SUCCESS == history.getStatus()) {
                            nbRecipientsSuccessfullySent += totalRecipients;
                        } else {
                            nbRecipientsFailureSent += totalRecipients;
                        }
                    }
```